### PR TITLE
Sort proficiencies and clean up percentage formatting

### PR DESCRIPTION
**Pull request by wapcaplet**
#### Summary

SUMMARY: Interface "Sort proficiencies and clean up percentage formatting"

#### Purpose of change

Proficiencies with names too long for the window width had their percentage of completion misaligned with other proficiencies with shorter names, due to using `trim_by_length` with too narrow a width. The name and percent could also run together, since no explicit space was included between them.

Proficiencies were also not correctly sorted by name.

#### Describe the solution

Right-align the percentages, and include a space between the name and percentage completion.

Swap the proficiency id and name in the sorting lists, so proficiencies will be sorted by name within the known/learning groups.

#### Describe alternatives you've considered

~~Another possibility is to keep the `4`, and insert a space before the percentage. That might look nicer.~~ Done.

#### Testing

Using an atomic lamp and debug hammerspace (and life support), craft a bunch of stuff to learn proficiencies. Check for nice alignment.

#### Additional context

Before:
image

After:
image

With scrollbar:
image

### CODE REVIEW

**Comment by anothersimulacrum** [comment]
I think that the extra space would look nicer than squishing them together, if you're willing to make that change.

**Comment by Kelenius** [comment]
Can you check that it works with proficiencies close to 100%? There's a save file at #44637.

File: src/player_display.cpp [hunk 1]
```diff
@@ -342,7 +342,7 @@ static void draw_proficiencies_tab( const catacurses::window &win, const unsigne
         if( !cur.known && cur.id->can_learn() ) {
             static_assert( grid_width == 26, "Reminder to update formatting"
                            "for this string when grid width changes" );
-            name = string_format( "%-22s%2.0f%%", trim_by_length( cur.id->name(), width - 4 ),
+            name = string_format( "%-22s%2.0f%%", trim_by_length( cur.id->name(), width - 3 ),
                                   cur.practice * 100 );
```
Comment (anothersimulacrum) [inline src/player_display.cpp:346]: This should resolve Kelenius's issue
```suggestion
                                  std::floor( cur.practice * 100 ) );
```

**Comment by Pupsi-Mupsi** [comment]
Can you check if it still works with an active scroll bar?

**Comment by wapcaplet** [comment]
Thanks for the feedback and review. I've revised it to have a space between name and percentage (compare "Principles of Trapping" before and after).

Verified that `std::floor` works for percentages close to 100% using save game from @Kelenius:

image

And checked with smaller window to see scrollbar - it looks OK:

image

**Pull request merged**